akka-stream/src/main/scala/akka/stream/RestartSettings.scala

```diff
@@ -44,6 +46,14 @@ final class RestartSettings private (
   def withMaxRestarts(count: Int, within: java.time.Duration): RestartSettings =
     copy(maxRestarts = count, maxRestartsWithin = within.asScala)
 
+  /** Scala API: Decides whether the failure should restart the stream */
+  def withRestartOn(restartOn: Throwable => Boolean): RestartSettings =
+    copy(restartOn = restartOn)
+
+  /** Java API: Decides whether the failure should restart the stream */
+  def withRestartOn(restartOn: java.util.function.Predicate[Throwable]): RestartSettings =
+    copy(restartOn = restartOn.asScala)
```

Comment:
I wonder if we really need separate APIs or if we couldn't have just this, can you check if a single method accepting `Predicate` wouldn't still be nicely usable from Scala (I'm a bit surprised that the current two methods are not ambigous when compiling Scala calling it).

Reply:
`Predicate` seems to work nicely from Scala